### PR TITLE
Adding schema constraints while creating JanusGraph schema

### PR DESCRIPTION
**Pull request by anki2189**
This pull request closes https://github.com/strongbox/strongbox/issues/1755

### CODE REVIEW

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java [hunk 1]
```diff
@@ -235,7 +236,83 @@ private void applySchemaChanges(JanusGraphManagement jgm)
...
+                                     "coordinates.name", "coordinates.build", "coordinates.abi", "coordinates.platform",
+                                     "coordinates.languageImplementationVersion", "coordinates.packaging",
+                                     "coordinates.distribution");
+
+        addVertexPropertyConstraints(jgm, ARTIFACT_TAG, "uuid");
+
+        addVertexPropertyConstraints(jgm, ARTIFACT_ID_GROUP, "uuid", "storageId", "repositoryId", "name");
+
+        addVertexPropertyConstraints(jgm, USER, "uuid", "username", "password",
+                                     "enabled", "roles", "securityTokenKey", "sourceId");
```
Comment (carlspring) [inline strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java]: Cosmetic: Could you please make all these one per line, where the list is longer than a line? Thanks! :)

Comment (anki2189) [reply strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java]: Fixed.

**Comment by anki2189** [comment]
@sbespalov Pls review the changes.

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java [hunk 2]
```diff
@@ -4,19 +4,30 @@
 {
 
     String ARTIFACT = "Artifact";
+
```
Comment (sbespalov) [inline strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java]: @anki2189 I would leave it without emty lines.

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java [hunk 3]
```diff
@@ -4,19 +4,30 @@
...
-    
+
     String ARTIFACT_TAG = "ArtifactTag";
+
     String ARTIFACT_ID_GROUP = "ArtifactIdGroup";
-    
+
     String USER = "User";
+
+    String TRANSACTION_PROPAGATION = "TransactionPropagationTestVertex";
```
Comment (sbespalov) [inline strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java]: @anki2189 AFAIK this is just for test, so we better use something else there and drop this `TransactionPropagationTestVertex` from schema.

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java [hunk 4]
```diff
@@ -235,7 +238,182 @@ private void applySchemaChanges(JanusGraphManagement jgm)
...
+
+        addVertexPropertyConstraints(jgm,
+                                     REMOTE_ARTIFACT,
+                                     "uuid",
+                                     "cached",
+                                     "created");
+
+        addVertexPropertyConstraints(jgm,
+                                     GENERIC_ARTIFACT_COORDINATES,
+                                     "uuid",
```
Comment (sbespalov) [inline strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java]: @anki2189 could you also please extract these property names into constants? Just like we have it with `Vertices` and `Edges`.

**Review by sbespalov — COMMENTED** [review]
@anki2189 many thanks for the PR!
could you please proceed with few comments here?

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java (hunk 2 above)
Comment (anki2189) [reply strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java]: Fixed.

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java (hunk 3 above)
Comment (anki2189) [reply strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/Vertices.java]: Changes done.

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java (hunk 4 above)
Comment (anki2189) [reply strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java]: Suggested changes are part of https://github.com/strongbox/strongbox-db/pull/20

**Comment by anki2189** [comment]
Build is passing all the tests https://github.com/strongbox/strongbox/pull/1758.

File: strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java (hunk 4 above)
Comment (sbespalov) [reply strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java]: @anki2189 would you mind to move it here? It seems that here it is more suitable than within `ArtifactArchiveListing`.

Comment (anki2189) [reply strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java]: @sbespalov  Changes pushed, pls verify.

**Comment by sbespalov** [comment]
merged!
thanks @anki2189 !

**Pull request merged**